src/os/ui/datetime/datecontrol.js

```diff
@@ -220,6 +254,20 @@ os.ui.datetime.DateControlCtrl.prototype.onDurationChanged = function() {
 };
 
 
+/**
+ * Set the duration to a date range relative to now.
+ *
+ * @param {number} days
+ * @private
+ */
+os.ui.datetime.DateControlCtrl.prototype.setRelativeDateRange = function(days) {
+  this['relativeDuration'] = true;
+  os.time.resetDate(this['startDate']);
```

Comment:
I think we want to initialize both dates to `new Date()` so it's actually using "now" versus the start of the current day, then adjust `startDate` by the number of days. After making that change, you should be able to remove `os.time.resetDate` as it will no longer be used.